packages/storage/amplify_storage_s3/example/integration_test/list_test.dart

```diff
@@ -20,32 +20,69 @@ void main() {
       '$uniquePrefix/file2.txt',
       '$uniquePrefix/subdir/file3.txt',
       '$uniquePrefix/subdir2#file4.txt',
+      '$uniquePrefix/file5.txt',
+      '$uniquePrefix/file6.txt',
+      '$uniquePrefix/subdir3/file7.txt',
+      '$uniquePrefix/subdir4#file8.txt',
     ];
     group('standard config', () {
+      final mainBucket =
+          StorageBucket.fromOutputs('Storage Integ Test main bucket');
+      final secondaryBucket = StorageBucket.fromOutputs(
+        'Storage Integ Test secondary bucket',
+      );
       setUpAll(() async {
         await configure(amplifyEnvironments['main']!);
-
-        for (final path in uploadedPaths) {
+        for (var pathIndex = 0;
+            pathIndex < uploadedPaths.length ~/ 2;
+            pathIndex++) {
           await Amplify.Storage.uploadData(
-            path: StoragePath.fromString(path),
+            path: StoragePath.fromString(uploadedPaths[pathIndex]),
             data: StorageDataPayload.bytes('test content'.codeUnits),
+            bucket: mainBucket,
+          ).result;
+        }
+        for (var pathIndex = uploadedPaths.length ~/ 2;
+            pathIndex < uploadedPaths.length;
+            pathIndex++) {
+          await Amplify.Storage.uploadData(
+            path: StoragePath.fromString(uploadedPaths[pathIndex]),
+            data: StorageDataPayload.bytes('test content'.codeUnits),
+            bucket: secondaryBucket,
           ).result;
         }
```

Comment:
Q: why break up the uploads?

Reply:
they are uploading to different buckets, I can put them in the same loop but I think two calls to upload will be required